File: Backend/SafalYatra/carOperator/getMyDrivers.php
<?php
include "../config/connection.php";
include "../utils/authHelper.php";


if (!isset($_POST['token'])) {
    echo json_encode(array(
        "success" => false,
        "message" => "Token is required"
    ));
    die();
}

$token = $_POST['token']; // Storing the token from POST request


$operatorId = getOperatorId($token);

if (!$operatorId) {
    echo json_encode(array(
        "success" => false,
        "message" => "Invalid token"
    ));
    die();
}


// Query to get the list of listDrivers
$query = "SELECT driver_id, operator_id, driver_name, phone_number, license_number, image_url, email, gender, age, experience, added_date, status FROM drivers 
 WHERE drivers.operator_id = '$operatorId';
";

$result = $connect->query($query);

// Check if there are listDrivers found
if ($result->num_rows > 0) {
    // Fetch the result as an associative array
    $MyDrivers = array();
    while ($row = $result->fetch_assoc()) {
        $MyDrivers[] = $row;
    }

    // Send the JSON response
    echo json_encode(array(
        "success" => true,
        "message" => "Drivers fetched successfully",
        "MyDrivers" => $MyDrivers
    ));
} else {
    echo json_encode(array(
        "success" => false,
        "message" => "Failed to fetch Drivers",

    ));
    die();
}

// Close the database connection
$connect->close();

File: Frontend/safal_yatra/lib/app/models/getMyDrivers.dart
// To parse this JSON data, do
//
//     final myDriversResponse = myDriversResponseFromJson(jsonString);

import 'dart:convert';

MyDriversResponse myDriversResponseFromJson(String str) =>
    MyDriversResponse.fromJson(json.decode(str));

String myDriversResponseToJson(MyDriversResponse data) =>
    json.encode(data.toJson());

class MyDriversResponse {
  final bool? success;
  final String? message;
  final List<MyDriver>? myDrivers;

  MyDriversResponse({
    this.success,
    this.message,
    this.myDrivers,
  });

  factory MyDriversResponse.fromJson(Map<String, dynamic> json) =>
      MyDriversResponse(
        success: json["success"],
        message: json["message"],
        myDrivers: json["MyDrivers"] == null
            ? []
            : List<MyDriver>.from(
                json["MyDrivers"]!.map((x) => MyDriver.fromJson(x))),
      );

  Map<String, dynamic> toJson() => {
        "success": success,
        "message": message,
        "MyDrivers": myDrivers == null
            ? []
            : List<dynamic>.from(myDrivers!.map((x) => x.toJson())),
      };
}

class MyDriver {
  final String? driverId;
  final String? operatorId;
  final String? driverName;
  final String? phoneNumber;
  final String? licenseNumber;
  final dynamic imageUrl;
  final String? email;
  final String? gender;
  final String? age;
  final String? experience;
  final DateTime? addedDate;
  final String? status;

  MyDriver({
    this.driverId,
    this.operatorId,
    this.driverName,
    this.phoneNumber,
    this.licenseNumber,
    this.imageUrl,
    this.email,
    this.gender,
    this.age,
    this.experience,
    this.addedDate,
    this.status,
  });

  factory MyDriver.fromJson(Map<String, dynamic> json) => MyDriver(
        driverId: json["driver_id"],
        operatorId: json["operator_id"],
        driverName: json["driver_name"],
        phoneNumber: json["phone_number"],
        licenseNumber: json["license_number"],
        imageUrl: json["image_url"],
        email: json["email"],
        gender: json["gender"],
        age: json["age"],
        experience: json["experience"],
        addedDate: json["added_date"] == null
            ? null
            : DateTime.parse(json["added_date"]),
        status: json["status"],
      );

  Map<String, dynamic> toJson() => {
        "driver_id": driverId,
        "operator_id": operatorId,
        "driver_name": driverName,
        "phone_number": phoneNumber,
        "license_number": licenseNumber,
        "image_url": imageUrl,
        "email": email,
        "gender": gender,
        "age": age,
        "experience": experience,
        "added_date": addedDate?.toIso8601String(),
        "status": status,
      };
}
